packages/components/inputs/src/base-input/BaseInput.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {
+  useEffect,
+  useCallback,
+  FocusEvent,
+  KeyboardEvent,
+  ChangeEvent,
+  useState,
+  ElementType,
+} from 'react';
+import { cx } from 'emotion';
+
+import {
+  usePrimitive,
+  PolymorphicComponentWithRef,
+  PolymorphicComponentProps,
+  PolymorphicComponent,
+} from '@contentful/f36-core';
+import getInputStyles from './BaseInput.styles';
+import { BaseInputInternalProps } from './types';
+
+const DEFAULT_TAG: ElementType = 'input';
+
+export type BaseInputProps<
+  E extends React.ElementType
+> = PolymorphicComponentProps<E, BaseInputInternalProps, 'disabled'>;
+
+const _BaseInput: PolymorphicComponentWithRef<
+  BaseInputInternalProps,
+  typeof DEFAULT_TAG
+> = (props, ref) => {
+  const {
+    as,
```

Comment:
```suggestion
    as = DEFAULT_TAG,
```
Otherwise it renders as a `div`